src/raster/rasterband.rs

```diff
@@ -863,6 +945,55 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: usize,
+    counts: HistCounts,
+}
+
+impl Histogram {
+    /// Histogram lower bound
+    pub fn min(&self) -> f64 {
+        self.min
+    }
+
+    /// Histogram upper bound
+    pub fn max(&self) -> f64 {
+        self.max
+    }
+    /// Histogram values for each bucket
+    pub fn counts(&self) -> &[i32] {
+        match &self.counts {
+            HistCounts::Borrowed(p) => unsafe { std::slice::from_raw_parts(*p, self.n_buckets) },
+            HistCounts::Owned(v) => v.as_slice(),
+        }
+    }
+
+    /// Histogram bucket size
+    pub fn bucket_size(&self) -> f64 {
+        (self.max - self.min) / self.counts().len() as f64
+    }
+}
+
+#[derive(Debug)]
```

Comment:
We probably want a manually-written `Debug` to handle the `Borrowed` case.

Reply:
Also, I wonder if we should turn this into a more generic thing. We have something similar -- `GcpRef`, but it looks quite differently. Probably badly-named, too.

Reply:
What do you want exactly? At the moment it shows that it is a pointer (wrapped on GdalAllocated). Do you want it to show a Vec?

Reply:
As a slice, you can delegate to that. It's much more useful than showing a pointer.